h/auth/policy.py

```diff
@@ -326,5 +348,19 @@ def _is_api_request(request):
             request.path not in ['/api/token', '/api/badge'])
 
 
+def _is_client_request(request):
+    """
+    Is client_auth authentication valid for the given request?
+
+    For initial rollout, authentication should be performed by
+    :py:class:`~h.auth.policy.AuthClientPolicy` only for requests
+    to the `POST /api/groups` endpoint.
+
+    This is intended to be temporary.
+    """
+    return (re.match(AUTH_TOKEN_PATH_PATTERN, request.path) and
+            request.method == 'POST')
```

Comment:
Can this be `request.matched_route.name == "api.groups" and request.method == "POST"` instead? That's the correct way to do it. The `/api/groups` path is served by the `"api.groups"` route:

https://github.com/hypothesis/h/blob/673189bb14e1151c5d5de77dce85ee8a067eeecf/h/routes.py#L110

And that route is only used [by the group read API](https://github.com/hypothesis/h/blob/673189bb14e1151c5d5de77dce85ee8a067eeecf/h/views/api/groups.py#L16-L19) (with `request_method='GET'`) and [by the group create API](https://github.com/hypothesis/h/blob/673189bb14e1151c5d5de77dce85ee8a067eeecf/h/views/api/groups.py#L39-L42) with `request_method='POST'`. This won't break if the API paths change but mostly it's simpler: no string pattern matching mistakes, no regex needed.

The group member APIs use a different route name, `"api.group_member"`, so they won't match:

https://github.com/hypothesis/h/blob/673189bb14e1151c5d5de77dce85ee8a067eeecf/h/routes.py#L113-L116

Reply:
In the tests [`pyramid_request.matched_route` is a mock](https://github.com/hypothesis/h/blob/673189bb14e1151c5d5de77dce85ee8a067eeecf/tests/h/conftest.py#L242) so test setup will have to do `pyramid_request.matched_route.name = "whatever"` to test it against different `matched_route` values.